File: lana/src/main/scala/com/financialforce/lana/cli/sfdx/SFDX.scala
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */
package com.financialforce.lana.cli.sfdx

import scala.concurrent.Future
import scala.scalajs.js
import scala.scalajs.js.annotation.JSGlobal

@js.native
@JSGlobal
class SFDXResponse[T <: js.Any](val status: Int, val result: T) extends js.Object

object SFDX {
  def apply(path: String, args: Seq[String]): Future[String] = Command(path, Seq("sfdx") ++ args)
}

File: src/sfdx/SFDX.ts
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */

import { Command } from "./Command";

export interface SFDXResponse<T> {
  status: number;
  result: T;
}

export class SFDX {
  static async apply(path: string, args: string[]): Promise<string> {
    return Command.apply(path, ["sfdx"].concat(args));
  }
}
